datafusion/physical-expr/src/intervals/cp_solver.rs

```diff
@@ -624,107 +473,15 @@ impl ExprIntervalGraph {
-fn propagate_time_interval_at_left(
-    left_child: &Interval,
-    right_child: &Interval,
-    parent: &Interval,
-    op: &Operator,
-    inverse_op: &Operator,
-) -> Result<Option<(Interval, Interval)>> {
-    // We check if the child's time interval(s) has a non-zero month or day field(s).
-    // If so, we return it as is without propagating. Otherwise, we first convert
-    // the time intervals to the `Duration` type, then propagate, and then convert
-    // the bounds to time intervals again.
-    let result = if let Some(duration) = convert_interval_type_to_duration(left_child) {
-        match apply_operator(inverse_op, parent, right_child)?.intersect(duration)? {
-            Some(value) => {
-                let left = convert_duration_type_to_interval(&value);
-                let right = propagate_right(&value, parent, right_child, op, inverse_op)?;
-                match (left, right) {
-                    (Some(left), Some(right)) => Some((left, right)),
-                    _ => None,
-                }
-            }
-            None => None,
-        }
-    } else {
-        propagate_right(left_child, parent, right_child, op, inverse_op)?
-            .map(|right| (left_child.clone(), right))
-    };
-    Ok(result)
-}
-
-/// During the propagation of [`Interval`] values on an [`ExprIntervalGraph`],
-/// if there exists a `timestamp - timestamp` operation, the result would be
-/// of type `Duration`. However, we may encounter a situation where a time interval
-/// is involved in an arithmetic operation with a `Duration` type. This function
-/// offers special handling for such cases, where the time interval resides on
-/// the right side of the operation.
-fn propagate_time_interval_at_right(
-    left_child: &Interval,
-    right_child: &Interval,
-    parent: &Interval,
-    op: &Operator,
-    inverse_op: &Operator,
-) -> Result<Option<(Interval, Interval)>> {
-    // We check if the child's time interval(s) has a non-zero month or day field(s).
-    // If so, we return it as is without propagating. Otherwise, we first convert
-    // the time intervals to the `Duration` type, then propagate, and then convert
-    // the bounds to time intervals again.
-    let result = if let Some(duration) = convert_interval_type_to_duration(right_child) {
-        match apply_operator(inverse_op, parent, &duration)?.intersect(left_child)? {
-            Some(value) => {
-                propagate_right(left_child, parent, &duration, op, inverse_op)?
-                    .and_then(|right| convert_duration_type_to_interval(&right))
-                    .map(|right| (value, right))
-            }
-            None => None,
-        }
-    } else {
-        apply_operator(inverse_op, parent, right_child)?
-            .intersect(left_child)?
-            .map(|value| (value, right_child.clone()))
-    };
-    Ok(result)
-}
-
-fn reverse_tuple<T, U>((first, second): (T, U)) -> (U, T) {
-    (second, first)
-}
-
 #[cfg(test)]
```

Comment:
is there a reason this entire file wasn't migrated over?

Reply:
I'm not sure all of them need to move to common, so I only move what I need for now.

Reply:
`ExprIntervalGraphNode` is not used for creating expr, but in execution plan